commit 54728e1dbad534190b571211cf932102fabdbf3c
Author: Scott Moser <smoser@brickies.net>
Date:   Thu Mar 7 19:33:05 2013 -0500

    changes to support dokucrypt plugin
    
    There are 2 changes here.
    * Resolving conflict with Linkwizard JS code (Editor Toolbar Issues)
      lib/scripts/linkwiz.js
    
      This solution makes almost everything work.
      The linkwizard is checking the textearea's form's pageid, but fails to
      do so, because moving the textarea out of the form is a critical part of
      the crypt's functionality
    
    * Resolving problems with empty pages and toolbar header buttons
      lib/scripts/edit.js
    
      The header buttons in the editor toolbar are not working.
      Reported by Termiter.

diff --git a/lib/scripts/edit.js b/lib/scripts/edit.js
index 5a5e829..c4892fa 100644
--- a/lib/scripts/edit.js
+++ b/lib/scripts/edit.js
@@ -146,7 +146,7 @@ function currentHeadlineLevel(textboxId){
     var field = jQuery('#' + textboxId)[0],
         s = false,
         opts = [field.value.substr(0,getSelection(field).start)];
-    if (field.form.prefix) {
+    if (field.form && field.form.prefix) { // DOKUCRYPT
         // we need to look in prefix context
         opts.push(field.form.prefix.value);
     }
diff --git a/lib/scripts/linkwiz.js b/lib/scripts/linkwiz.js
index c55650d..3ed081c 100644
--- a/lib/scripts/linkwiz.js
+++ b/lib/scripts/linkwiz.js
@@ -44,6 +44,7 @@ var dw_linkwiz = {
                .appendTo('.dokuwiki:first');
 
         dw_linkwiz.textArea = $editor[0];
+        dw_linkwiz.formid = dw_linkwiz.textArea.form['id'].value; // DOKUCRYPT
         dw_linkwiz.result = jQuery('#link__wiz_result')[0];
 
         // scrollview correction on arrow up/down gets easier
@@ -230,7 +231,7 @@ var dw_linkwiz = {
         }
 
         // prepend colon inside namespaces for non namespace pages
-        if(dw_linkwiz.textArea.form.id.value.indexOf(':') != -1 &&
+        if(dw_linkwiz.formid.indexOf(':') != -1 &&   // DOKUCRYPT
            link.indexOf(':') == -1){
            link = ':' + link;
         }
